devdoc/clds_hash_table_requirements.md

```diff
@@ -7,9 +7,17 @@
 The module provides the following functionality:
 - Inserting items in the hash table
 - Delete an item from the hash table by its key
+- Replace an item in the hash table by its key
+- Find an item in the hash table by its key
 
 All operations can be concurrent with other operations of the same or different kind.
 
+This hash table supports taking a snapshot of the current state by blocking all changes to the table and dumping the nodes.
+
+### Future work
+
+The snapshot functionality will be extended in the future so that concurrent operations are possible. This will be done by storing changes during snapshots in a separate structure and then merging after the snapshot completes.
```

Comment:
>This will be done by storing changes during snapshots in a separate structure and then merging after the snapshot completes. [](start = 102, length = 124)

Love.